### PR TITLE
Fix stuck volley tests

### PR DESCRIPTION
**Pull request by laurit**
Based on https://github.com/signalfx/splunk-otel-android/pull/230
Instead of trying to make `Looper` behave this pr uses `Executor` for volley tests. A helper class for dumping thread stacks when a test gets stuck is also added.
@Enkelian could you review

### CODE REVIEW

File: splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java [hunk 1]
```diff
@@ -190,17 +184,19 @@ public void reusedRequest() throws IOException {
...
+        }
+        TestResponseListener testResponseListener = new TestResponseListener();
+        StringRequest stringRequest = new StringRequest(Request.Method.GET, url.toString(), testResponseListener, error -> {});
         testQueue.addToQueue(stringRequest);
         testQueue.addToQueue(stringRequest);
 
-        Scheduler scheduler = shadowOf(getMainLooper()).getScheduler();
-        while (!scheduler.advanceToLastPostedRunnable());
-        while (!scheduler.advanceToNextPostedRunnable());
+        testResponseListener.countDownLatch.await();
```
Comment (mateuszrzeszutek) [inline splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java:199]: Nit
```suggestion
        testResponseListener.countDownLatch.await(10, TimeUnit.SECONDS);
```

File: splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java [hunk 2]
```diff
@@ -214,10 +210,60 @@ public void reusedRequest() throws IOException {
...
+                        throw new AssertionError(e);
+                    }
+                    StringRequest stringRequest = new StringRequest(Request.Method.GET, url.toString(), testResponseListener, error -> {});
+                    testQueue.addToQueue(stringRequest);
+                };
+            pool.submit(job);
+        }
+
+        latch.countDown();
+        testResponseListener.countDownLatch.await();
```
Comment (mateuszrzeszutek) [inline splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java:251]: Nit:
```suggestion
        testResponseListener.countDownLatch.await(10, TimeUnit.SECONDS);
```

File: splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java [hunk 3]
```diff
@@ -214,10 +210,60 @@ public void reusedRequest() throws IOException {
...
+            server.enqueue(new MockResponse().setBody(responseBody));
+        }
+
+        server.play();
+        URL url = server.getUrl("/success");
+
+        CountDownLatch latch = new CountDownLatch(1);
+        ExecutorService pool = Executors.newFixedThreadPool(4);
+
+        class TestResponseListener implements Response.Listener<String> {
```
Comment (mateuszrzeszutek) [inline splunk-otel-android-volley/src/test/java/com/splunk/rum/TracingHurlStackTest.java:228]: This is almost the same as in the other test, WDYT about pulling this up to the class level?

**Review by mateuszrzeszutek — APPROVED** [review]
🚀

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by Enkelian — APPROVED** [review]
I think it's better to go ahead with this PR rather than https://github.com/signalfx/splunk-otel-android/pull/230/files since it removes the usage of deprecated Robolectric API

**Pull request merged**